gax/src/test/java/com/google/api/gax/rpc/ClientContextTest.java

```diff
@@ -190,7 +192,7 @@ public String getTransportName() {
 
     @Override
     public boolean needsCredentials() {
-      return credentials == null;
+      return credentials == null && !headers.containsKey("x-goog-api-key");
```

Comment:
Why did need this? It seems like it masks credential != null check

Reply:
If header contains `x-goog-api-key`, then we don't need credentials. We cannot use both API key and credentials tokens.

Reply:
Yes, but this is a fake provider and if this check matters here, why we don't setup provider similarly outside of tests?

Reply:
The test throws an exception in `getTransportChannel` if `needsCredentials`, so the right place to check this is in `getTransportChannel`. Updated the `getTransportChannel` to return the fake transport channel if api key is in the header.